File: 3379. Transformed Array/constructTransformedArray.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var constructTransformedArray = function (nums) {
    const len = nums.length
    let res = new Array(len).fill(0)

    for (let i = 0; i < len; i++) {
        let val = nums[i]
        if (val !== 0) {
            const idx = ((i + val) % len + len) % len
            res[i] = nums[idx]
        }
    }
    return res;
};

File: 3379. Transformed Array/constructTransformedArray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        int sz = nums.size();
        vector<int> res(sz,0);

        for(int i=0; i<sz; i++){
            int val = nums[i];
            if(nums[i]!= 0){
                int idx = (((i+val) % sz) + sz)% sz;
                res[i] = nums[idx];
            }
        }
        return res;
    }
};